controllers/grid/eventLog/SubmissionEventLogGridHandler.inc.php

```diff
@@ -72,7 +75,21 @@ function setSubmission($submission) {
 	function authorize($request, &$args, $roleAssignments) {
 		import('lib.pkp.classes.security.authorization.SubmissionAccessPolicy');
 		$this->addPolicy(new SubmissionAccessPolicy($request, $args, $roleAssignments));
-		return parent::authorize($request, $args, $roleAssignments);
+
+		$success = parent::authorize($request, $args, $roleAssignments);
+
+		// Prevent authors from accessing review details, even if they are also
+		// assigned as an editor, sub-editor or assistant.
+		$userAssignedRoles = $this->getAuthorizedContextObject(ASSOC_TYPE_ACCESSIBLE_WORKFLOW_STAGES);
```

Comment:
One more comment: I've just tested it when I am assigned as editor (manager role) and author, the history is not anonymized, because: here there will be no $userAssignedRoles because in the SubmissionAccessPolicy the rule for manger will immediately be applied -- that they have access to everything. Thus additionally to the SubmissionAccessPolicy above, maybe the following would also be needed:
```
import('lib.pkp.classes.security.authorization.internal.UserAccessibleWorkflowStageRequiredPolicy');
$this->addPolicy(new UserAccessibleWorkflowStageRequiredPolicy($request, WORKFLOW_TYPE_EDITORIAL));
```
where that array $userAssignedRoles will be filled...

Reply:
Oh I just noticed that I didn't address this! Arg! I'm going to go ahead and merge for the translators, and then fix this in a separate commit.